src/IB/BrinkmanAdvDiffBcHelper.cpp

```diff
@@ -0,0 +1,974 @@
+    for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+    {
+        Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+        {
+            Pointer<Patch<NDIM> > patch = level->getPatch(p());
+            const Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+            const Box<NDIM>& patch_box = patch->getBox();
+            const double* patch_dx = patch_geom->getDx();
+            double vol_cell = 1.0;
+            for (int d = 0; d < NDIM; ++d) vol_cell *= patch_dx[d];
+
+            Pointer<CellData<NDIM, double> > F_data = patch->getPatchData(F_idx);
+
+            for (Box<NDIM>::Iterator it(patch_box); it; it++)
+            {
+                CellIndex<NDIM> ci(it());
+
+                // Loop over all the level sets and add up their contributions
+                // TODO: Pointwise check that level sets don't overlap
+                double brinkman_forcing = 0.0;
+                for (const auto& bc_prop : brinkman_zones)
+                {
+                    // Get the BC specifications for each zone
+                    Pointer<CellVariable<NDIM, double> > ls_solid_var = bc_prop.ls_solid_var;
+                    AdvDiffBrinkmanPenalizationBcType bc_type = bc_prop.bc_type;
+                    double bc_val = bc_prop.bc_val;
+                    double eta = bc_prop.eta;
+
+                    bool requires_callbacks = bc_prop.callback != nullptr;
+                    double num_interface_cells = bc_prop.num_interface_cells;
+                    const double alpha = num_interface_cells * std::pow(vol_cell, 1.0 / static_cast<double>(NDIM));
+
+                    if (requires_callbacks) continue;
+#if !defined(NDEBUG)
+                    TBOX_ASSERT(bc_type == DIRICHLET);
+                    TBOX_ASSERT(!requires_callbacks);
+#endif
+                    const int phi_idx =
+                        var_db->mapVariableAndContextToIndex(ls_solid_var, d_adv_diff_solver->getNewContext());
+                    Pointer<CellData<NDIM, double> > ls_solid_data = patch->getPatchData(phi_idx);
+                    double phi = (*ls_solid_data)(ci);
+
+                    IndicatorFunctionType indicator_func_type = bc_prop.indicator_func_type;
+                    double Hphi = 0.0;
+                    if (indicator_func_type == SMOOTH)
+                        Hphi = IBTK::smooth_heaviside(phi, alpha);
+                    else if (indicator_func_type == DISCONTINUOUS)
+                        Hphi = IBTK::discontinuous_heaviside(phi);
+
+                    // Note: assumes that chi is positive when phi is negative
+                    const double chi = (1.0 - Hphi);
+
+                    // Homogeneous Neumann BCs have no contribution to the forcing term
+                    if (bc_type == DIRICHLET)
+                    {
+                        brinkman_forcing += (chi / eta * bc_val);
```

Comment:
@Raamkrishnan : for better readability use parentheses at the correct spot 
`brinkman_forcing += (chi / eta) * bc_val;`